zerver/actions/create_user.py

```diff
@@ -317,7 +317,7 @@ def notify_created_user(user_profile: UserProfile) -> None:
         send_event(user_profile.realm, event, user_ids_with_real_email_access)
 
     if user_ids_without_real_email_access:
-        del person["delivery_email"]
+        person["delivery_email"] = None
```

Comment:
It probably would be a nice future refactor to just call `format_user_row` twice with different parameters, rather than doing this surgery. I'm pretty sure that function doesn't have any real cost, and one wouldn't have to think about this surgery too closely. But I don't think that's a blocker for merging this.

Reply:
Will look at this.